datacollectiondemo/source/java/src/org/kaaproject/kaa/demo/datacollection/DataCollectionDemo.java

```diff
@@ -60,54 +64,98 @@ public void onStopped() {
 
         // Set record count strategy for uploading logs with count threshold set to 1.
         // Defined strategy configuration informs to upload every log record as soon as it is created.
-        // The default strategy uploads logs after either a threshold logs count
-        // or a threshold logs size has been reached.
-        kaaClient.setLogUploadStrategy(new RecordCountLogUploadStrategy(1));
+        // Use "WithTimeLimit" modification so that application will not wait for infinite time,
+        // if current "logs count threshold" will not be submultiple of real logs count
+        kaaClient.setLogUploadStrategy(new RecordCountWithTimeLimitLogUploadStrategy(LOGS_DEFAULT_THRESHOLD, LOGS_MAX_WAIT_TIME, TimeUnit.SECONDS));
+
+        //Add a listener to display endpoint's profile/configuration data each time configuration is updated.
+        kaaClient.addConfigurationListener(new ConfigurationListener() {
+            @Override
+            public void onConfigurationUpdate(LogUploadThreshold configuration) {
+                LOG.info("Endpoint configuration was updated.");
+                displayConfiguration(configuration);
+
+                Integer threshold = configuration.getThreshold();
+                if (threshold != null) {
+                    if (threshold > 0) {
+                        kaaClient.setLogUploadStrategy(new RecordCountWithTimeLimitLogUploadStrategy(threshold, LOGS_MAX_WAIT_TIME, TimeUnit.SECONDS));
+                        LOG.warn("Set new logs count threshold value = {}.", threshold);
+                    } else {
+                        LOG.warn("Threshold value (= {}) in updated configuration is less than 1 so ignore it.", threshold);
+                    }
+                    LOG.info("--= Input \"Y\" to repeat data sending or other symbol to exit demo application =--", LOGS_DEFAULT_THRESHOLD);
+                }
```

Comment:
if (threshold == null) {
return;
}
 and other code to first level. It's good to read